editor/plugins/script_editor_plugin.cpp

```diff
@@ -1236,14 +1237,15 @@ void ScriptEditor::_menu_option(int p_option) {
 			_update_script_names();
 		} break;
 		case TOGGLE_SCRIPTS_PANEL: {
+			toggle_scripts_panel();
 			if (current) {
-				ScriptTextEditor *editor = Object::cast_to<ScriptTextEditor>(current);
-				toggle_scripts_panel();
-				if (editor) {
-					editor->update_toggle_scripts_button();
-				}
+				current->update_toggle_scripts_button();
 			} else {
-				toggle_scripts_panel();
+				Node *tab = tab_container->get_child(tab_container->get_current_tab());
```

Comment:
This one might work in this specific case but in general not all children of `TabContainer` are treated as tabs (only non top-level Controls are considered tabs) and thus it's not guaranteed that tab index matches index in parent. To obtain tab control by tab index `TabContainer::get_tab_control(int tab_index)` should be used. Or specifically for the current tab:
```suggestion
				Node *tab = tab_container->get_current_tab_control();
```